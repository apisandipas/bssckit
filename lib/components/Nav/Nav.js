"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _base = require("../Utilities/base");

var _theme = _interopRequireDefault(require("../../theme"));

var _themeFunctions = require("../../utils/themeFunctions");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject13() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  flex-wrap: wrap;\n  padding-left: ", ";\n  margin-bottom: ", ";\n  list-style: none;\n  & > a {\n    text-decoration: none;\n    ", "\n    ", "\n    ", "\n    ", "\n  }\n  &:last-child {\n   flex: 1;\n  }\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n"]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteral(["\n      @media (max-width: ", ") {\n        display: none;\n        flex-basis: auto;\n        flex-direction: column;\n      }\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteral(["\n      @media (max-width: ", ") {\n        display: flex;\n        flex-basis: auto;\n        flex-direction: column;\n        & > a {\n          padding-right: ", ";\n          padding-left: ", ";\n          flex: 1;\n          text-align: left;\n        }\n      }\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteral(["\n    & > a {\n      margin-bottom: -1px;\n    }\n  "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["\n    border-bottom: ", " ", ";\n  "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n    flex: 1 1 auto;\n  "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n      text-align: center;\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n    flex-grow: 1;\n  "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    flex-basis: 0;\n  "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    flex-direction: column !important;\n  "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n      justify-content: flex-start !important;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n      justify-content: flex-end !important;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n      justify-content: center !important;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var justifyContent = function justifyContent(props) {
  if (props.center) {
    return (0, _styledComponents.css)(_templateObject());
  } else if (props.end) {
    return (0, _styledComponents.css)(_templateObject2());
  } else if (props.start) {
    return (0, _styledComponents.css)(_templateObject3());
  }

  return '';
};

var flexDirection = function flexDirection(props) {
  return props.vertical && (0, _styledComponents.css)(_templateObject4());
};

var flexBasis = function flexBasis(props) {
  return props.justified && (0, _styledComponents.css)(_templateObject5());
};

var flexGrow = function flexGrow(props) {
  return props.justified && (0, _styledComponents.css)(_templateObject6());
};

var textAlign = function textAlign(props) {
  if (props.justified || props.fill) {
    return (0, _styledComponents.css)(_templateObject7());
  }

  return '';
};

var flex = function flex(props) {
  return props.fill && (0, _styledComponents.css)(_templateObject8());
};

var border = function border(props) {
  return props.tabs && (0, _styledComponents.css)(_templateObject9(), (0, _themeFunctions.getBorder)(props, 'nav', 'default'), (0, _themeFunctions.getColor)(props, 'nav', 'borderColorTabs'));
};

var margin = function margin(props) {
  return props.tabs && (0, _styledComponents.css)(_templateObject10());
};

var navCollapse = function navCollapse(props) {
  if ((props.expandSm || props.expandMd || props.expandLg || props.expandXl) && props.collapse && !props.hidden) {
    return (0, _styledComponents.css)(_templateObject11(), (0, _themeFunctions.getBreakpointSize)(props), (0, _themeFunctions.getPadding)(props, 'nav', 'collapsNoteHiddenARight'), (0, _themeFunctions.getPadding)(props, 'nav', 'collapsNoteHiddenALeft'));
  } else if ((props.expandSm || props.expandMd || props.expandLg || props.expandXl) && props.collapse && props.hidden) {
    return (0, _styledComponents.css)(_templateObject12(), (0, _themeFunctions.getBreakpointSize)(props));
  }

  return '';
};

var Nav = (0, _styledComponents["default"])(_base.Nav)(_templateObject13(), function (props) {
  return (0, _themeFunctions.getPadding)(props, 'nav', 'left');
}, function (props) {
  return (0, _themeFunctions.getMargin)(props, 'nav', 'bottom');
}, function (props) {
  return flex(props);
}, function (props) {
  return flexBasis(props);
}, function (props) {
  return flexGrow(props);
}, function (props) {
  return textAlign(props);
}, function (props) {
  return justifyContent(props);
}, function (props) {
  return flexDirection(props);
}, function (props) {
  return border(props);
}, function (props) {
  return margin(props);
}, function (props) {
  return navCollapse(props);
});
Nav.defaultProps = {
  theme: _theme["default"]
};
var _default = Nav;
exports["default"] = _default;