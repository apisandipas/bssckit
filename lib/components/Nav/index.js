"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Nav", {
  enumerable: true,
  get: function get() {
    return _Nav["default"];
  }
});
Object.defineProperty(exports, "NavLink", {
  enumerable: true,
  get: function get() {
    return _NavLink["default"];
  }
});

var _Nav = _interopRequireDefault(require("./Nav"));

var _NavLink = _interopRequireDefault(require("./NavLink"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }