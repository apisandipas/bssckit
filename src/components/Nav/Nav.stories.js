import React from 'react';
import { Nav, NavLink } from './index';

export default {
  title: 'Nav',
};

export const simple = () => (
  <Nav>
    <NavLink href='#' active>
      Active
    </NavLink>
    <NavLink href='#'>Link</NavLink>
    <NavLink href='#'>Link</NavLink>
    <NavLink href='#' disabled>
      Disabled
    </NavLink>
  </Nav>
);

export const pulled = () => (
  <Nav>
    <NavLink href='#'>Link</NavLink>
    <NavLink href='#'>Link</NavLink>
    <NavLink href='#'>Link</NavLink>
    <NavLink href='#' mlAuto>
      Pulled Link
    </NavLink>
    <NavLink href='#'>Pulled link</NavLink>
  </Nav>
);

export const centered = () => (
  <Nav center>
    <NavLink href='#' active>
      Active
    </NavLink>
    <NavLink href='#'>Link</NavLink>
    <NavLink href='#'>Link</NavLink>
    <NavLink href='#' disabled>
      Disabled
    </NavLink>
  </Nav>
);

export const aligned = () => (
  <Nav end>
    <NavLink href='#' active>
      Active
    </NavLink>
    <NavLink href='#'>Link</NavLink>
    <NavLink href='#'>Link</NavLink>
    <NavLink href='#' disabled>
      Disabled
    </NavLink>
  </Nav>
);

export const tabs = () => (
  <Nav tabs>
    <NavLink tabs href='#' active>
      Active
    </NavLink>
    <NavLink tabs href='#'>
      Link
    </NavLink>
    <NavLink tabs href='#'>
      Link
    </NavLink>
    <NavLink tabs href='#' disabled>
      Disabled
    </NavLink>
  </Nav>
);

export const pills = () => (
  <Nav tabs>
    <NavLink pills href='#' active>
      Active
    </NavLink>
    <NavLink pills href='#'>
      Link
    </NavLink>
    <NavLink pills href='#'>
      Link
    </NavLink>
    <NavLink pills href='#' disabled>
      Disabled
    </NavLink>
  </Nav>
);

export const justified = () => (
  <Nav justified>
    <NavLink pills item href='#' active>
      Active
    </NavLink>
    <NavLink pills item href='#'>
      Link
    </NavLink>
    <NavLink pills item href='#'>
      Link
    </NavLink>
    <NavLink pills item href='#' disabled>
      Disabled
    </NavLink>
  </Nav>
);

export const fill = () => (
  <Nav fill>
    <NavLink pills item href='#' active>
      Active
    </NavLink>
    <NavLink pills item href='#'>
      Link
    </NavLink>
    <NavLink pills item href='#'>
      Link
    </NavLink>
    <NavLink pills item href='#' disabled>
      Disabled
    </NavLink>
  </Nav>
);

export const vertical = () => (
  <Nav vertical>
    <NavLink pills item href='#' active>
      Active
    </NavLink>
    <NavLink pills item href='#'>
      Link
    </NavLink>
    <NavLink pills item href='#'>
      Link
    </NavLink>
    <NavLink pills item href='#' disabled>
      Disabled
    </NavLink>
  </Nav>
);
