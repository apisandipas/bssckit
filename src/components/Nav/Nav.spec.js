import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';

import Nav from './Nav';

describe('Nav component', () => {
  it('matches Nav', () => {
    const { container } = render(<Nav />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('display', 'flex');
    expect(container.firstChild).toHaveStyleRule('flex-wrap', 'wrap');
    expect(container.firstChild).toHaveStyleRule('padding-left', '0');
    expect(container.firstChild).toHaveStyleRule('margin-bottom', '0');
    expect(container.firstChild).toHaveStyleRule('list-style', 'none');
    expect(container.firstChild).toHaveStyleRule('text-decoration', 'none', {
      modifier: '& > a',
    });
    expect(container.firstChild).toHaveStyleRule('flex', '1', {
      modifier: '&:last-child',
    });
  });

  it('matches Nav center', () => {
    const { container } = render(<Nav center />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('justify-content', 'center !important');
  });

  it('matches Nav end', () => {
    const { container } = render(<Nav end />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('justify-content', 'flex-end !important');
  });

  it('matches Nav start', () => {
    const { container } = render(<Nav start />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('justify-content', 'flex-start !important');
  });

  it('matches Nav fill', () => {
    const { container } = render(<Nav fill />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('flex', '1 1 auto', {
      modifier: '& > a',
    });
    expect(container.firstChild).toHaveStyleRule('text-align', 'center', {
      modifier: '& > a',
    });
  });

  it('matches Nav justified', () => {
    const { container } = render(<Nav justified />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('flex-basis', '0', {
      modifier: '& > a',
    });
    expect(container.firstChild).toHaveStyleRule('flex-grow', '1', {
      modifier: '& > a',
    });
    expect(container.firstChild).toHaveStyleRule('text-align', 'center', {
      modifier: '& > a',
    });
  });

  it('matches Nav tabs', () => {
    const { container } = render(<Nav tabs />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border-bottom', '1px solid #cccecf');
    expect(container.firstChild).toHaveStyleRule('margin-bottom', '-1px', {
      modifier: '& > a',
    });
  });

  it('matches Nav vertical', () => {
    const { container } = render(<Nav vertical />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('flex-direction', 'column !important');
  });
});
