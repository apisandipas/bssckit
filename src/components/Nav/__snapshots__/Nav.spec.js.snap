// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches Nav 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
}

.c0 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<nav
  class="sc-fznKkj c0"
/>
`;

exports[`Nav component matches Nav center 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
  -webkit-box-pack: center !important;
  -webkit-justify-content: center !important;
  -ms-flex-pack: center !important;
  justify-content: center !important;
}

.c0 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<nav
  class="sc-fznKkj c0"
/>
`;

exports[`Nav component matches Nav end 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
  -webkit-box-pack: end !important;
  -webkit-justify-content: flex-end !important;
  -ms-flex-pack: end !important;
  justify-content: flex-end !important;
}

.c0 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<nav
  class="sc-fznKkj c0"
/>
`;

exports[`Nav component matches Nav fill 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
}

.c0 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  text-align: center;
}

.c0:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<nav
  class="sc-fznKkj c0"
/>
`;

exports[`Nav component matches Nav justified 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
}

.c0 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: center;
}

.c0:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<nav
  class="sc-fznKkj c0"
/>
`;

exports[`Nav component matches Nav start 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
  -webkit-box-pack: start !important;
  -webkit-justify-content: flex-start !important;
  -ms-flex-pack: start !important;
  justify-content: flex-start !important;
}

.c0 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<nav
  class="sc-fznKkj c0"
/>
`;

exports[`Nav component matches Nav tabs 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
  border-bottom: 1px solid #cccecf;
}

.c0 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 > a {
  margin-bottom: -1px;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<nav
  class="sc-fznKkj c0"
/>
`;

exports[`Nav component matches Nav vertical 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
  -webkit-flex-direction: column !important;
  -ms-flex-direction: column !important;
  flex-direction: column !important;
}

.c0 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<nav
  class="sc-fznKkj c0"
/>
`;
